tools/sfxconvert.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+
+AudioFunctions = {}
+Verbose = False
+Verbose2 = False
+
+def set_verbose(v):
+    global Verbose
+    global Verbose2
+
+    Verbose = v
+    Verbose2 = v
+    return 1
+
+def make_audio_dict(AudioFunctions, repo):
+    with open(repo + os.sep + 'include' + os.sep + 'functions.h','r') as funcfile:
+        funcdata = funcfile.readlines()
+    for i, line in enumerate(funcdata):
+        if(line.count('sfxId')):
+            funcname, argnum = get_func_data(funcdata,i)
+            if(funcname != None):
+                AudioFunctions[funcname] = argnum
+    return 1
+
+def get_func_data(funcdata,i):
+    j = i
+    while(funcdata[j - 1].count(';') == 0):
+        j -= 1
+    k = i + 1
+    while(funcdata[k - 1].count(';') == 0):
+        k += 1
+    prototype = "".join(funcdata[j:k])
+    argdata = prototype.replace('(',',').replace(')',',').split(',')
+    funcname = argdata[0].split(' ')[1]
+    for x in range(len(argdata)):
+        if(argdata[x].count('sfxId')):
+            break;
+    if(x == len(argdata) - 1):
+        print('sfxId not found in ', funcname)
+        return None,-1
+    return funcname, x - 1
+
+def lookup_sfx(idnum, repo):
+    if(type(idnum) is int):
+        id = '0x' + format(id,'X')
+    else:
+        id = idnum
+    id,sfxFlag = fix_sfx_flag(id)
+    with open(repo + os.sep + 'include' + os.sep + 'sfx.h','r') as sfxfile:
+        for line in sfxfile:
+            if(line.count(id)):
+                return line.split(' ')[1] + sfxFlag
+    return 'INVALID_ID'
+
+def fix_sfx_flag(id):
+    if(id.endswith(' - SFX_FLAG')):
+        splitdata = id.split('-')
+        return splitdata[0].strip(), ' -' + splitdata[1]
+    if not(int(id,16) & 0x800):
+        newid = '0x' + format(int(id,16) + 0x800,'X')
+        sfxFlag = ' - SFX_FLAG'
+    else :
+        newid = id
+        sfxFlag = ''
+    return newid,sfxFlag
+
+def fix_sfx_func(sourcedata, i, j, repo):
+    data = ''.join(sourcedata[i:j])
+    func = find_audio_func(data)
+    if(not func):
+        print('Function parse error at line', i)
+        return -3
+    index = data.find(func)
+    argnum = AudioFunctions.get(func,-1)
+    if(argnum == -1 or index == -1):
+        print('Function lookup error at line', i, 'in', func)
+        return -1
+    args = data[index:].replace('(',',').replace(')',',').split(',')
+    sfxId = args[argnum + 1].strip()
+    if(sfxId.count('0x') == 0):
+        return 0
```

Comment:
If I'm reading this correctly, it doesn't handle decimal values? I'd probably suggest handling them, but it may not be worth so I'll leave it up to you

Reply:
Eh... the purpose of that check is to make sure I'm not going to try to replace an SFX id that's already been processed. mips2c outputs hex by default, so I don't think needing decimal is going to be a common use case. Also the program description specifies the values should be in hex.

Reply:
you can check for a string to see if you should be replacing

Reply:
Oh I see. But there are actually a bunch of sfx defines with 0x in them, so the current check is probably not enough

Reply:
Whoa you're right. I didn't know about those. This would cause a fatal error when it tries to feed that into fix_sfx_flag. I'll need to rethink that check

Reply:
OK, new error detection. Checks for the string `NA_SE` to consider an argument already converted and will now work on purely decimal SFX ids.